src/utils/utils.jl

```diff
@@ -498,13 +478,13 @@ function transform_array_for_hdf(
     return t_lin_cost
 end
 
-function transform_array_for_hdf(data::Vector{QuadraticFunctionData})
-    all(get_constant_term.(data) .== 0) ||
-        throw(ArgumentError("Not yet implemented for nonzero constant term"))
-    data = [(get_quadratic_term(q), (get_proportional_term(q))) for q in data]
+function transform_array_for_hdf(
```

Comment:
You might want to implement this function closer to the previous method. Even when there is code repetition some of the variables' types won't be determined at run time.